CMP-3777: Enhance TestManualRulesTailoredProfile to cover 61342 logic

xiaojiey opened the pull request:
what's in the PR:
    Split manual and enabled rules tests into separate test cases
    
    Address review feedback to improve test organization:
    - Rename TestManualRulesTailoredProfile to TestTailoringManualRulesDoesNotGenerateRemediations
    - Extract enabled rules testing into new test: TestTailoringEnabledRulesGenerateRemediations
    - Update variable names from "regular" to "enabled" for clarity
    - Add clear comments that the rule is expected to fail
    - Make the two tests complementary for better readability
    
    The manual rules test verifies that rules added via ManualRules:
    - Report as MANUAL status
    - Do NOT generate remediations
    
    The enabled rules test verifies that rules added via EnableRules:
    - Report as FAIL (not MANUAL) when they fail
    - DO generate remediations when they fail

xiaojiey commented on tests/e2e/parallel/main_test.go:
```diff
@@ -4778,6 +4778,103 @@ func TestManualRulesTailoredProfile(t *testing.T) {
 		t.Fatal("expected no remediation")
 	}
+
+	// Create a TailoredProfile with the same rule as EnableRules (not ManualRules)
```
Yes, correct. That's why I think this PR could be optional as the key function MANUAL rules already exits.

xiaojiey commented on tests/e2e/parallel/main_test.go:
```diff
@@ -4778,6 +4778,103 @@ func TestManualRulesTailoredProfile(t *testing.T) {
+	// For regular rules, remediations may be created (if the check fails)
+	// This is the key difference from manual rules which should never create remediations
+	if len(regularRemList.Items) == 0 {
+		t.Log("Warning: Expected remediation for failed check in regular profile, but none found")
```
Yes. The purpose is to show that the same rule produces different results in ManualRules and EnableRules. ManualRules is the main function, and EnableRules is only for comparison. Since the EnableRules check is optional, I can close the PR if you think it’s unnecessary.

xiaojiey commented:
/retest

xiaojiey commented:
/retest

rhmdnd reviewed (changes requested):
I think this is good test coverage to add 🎉 .

My only real recommendation is that we break it out into its own test (given the length of the existing test). Otherwise, it's very readable.

Thanks!

xiaojiey committed:
Split manual and enabled rules tests into separate test cases

Address review feedback to improve test organization:
- Rename TestManualRulesTailoredProfile to TestTailoringManualRulesDoesNotGenerateRemediations
- Extract enabled rules testing into new test: TestTailoringEnabledRulesGenerateRemediations
- Update variable names from "regular" to "enabled" for clarity
- Add clear comments that the rule is expected to fail
- Make the two tests complementary for better readability

The manual rules test verifies that rules added via ManualRules:
- Report as MANUAL status
- Do NOT generate remediations

The enabled rules test verifies that rules added via EnableRules:
- Report as FAIL (not MANUAL) when they fail
- DO generate remediations when they fail

Anna-Koudelkova commented on tests/e2e/parallel/main_test.go:
Not sure how Xiaojie has resolved this, but seems like this test case is now passing in the parallel run

vickeybrown reviewed (approved):
Failing parallel test seems to be due to test flakeyness not related to PR
